File: 896-monotonic-array/896-monotonic-array.java
class Solution {
    public boolean isMonotonic(int[] nums) {
        boolean isNonIncreasing = true;
        boolean isNonDecreasing = true;
        
        for(int i=1; i<nums.length; i++) {
            int difference = nums[i]-nums[i-1];
            
            if(difference >0) isNonIncreasing = false;
            if(difference <0) isNonDecreasing = false;
            
        }
        
        return isNonIncreasing || isNonDecreasing;
            
        
        
    }
}

File: 896-monotonic-array/896-monotonic-array.py
class Solution:
    def isMonotonic(self, nums: List[int]) -> bool:
        isNonDecreasing = True
        isNonIncreasing = True
        
        for i in range(1,len(nums)):
            difference = nums[i] - nums[i-1]
            
            if difference >0:
                isNonIncreasing =False
            
            if difference <0:
                isNonDecreasing =False
                
        
        return isNonIncreasing or isNonDecreasing